### PR TITLE
Add a dev_mode to TestSuiteCore

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: e3/testsuite/__init__.py [hunk 1]
```diff
@@ -97,6 +97,11 @@ def __init__(self, root_dir):
         self.test_counter = 0
         self.test_status_counters = {s: 0 for s in TestStatus}
 
+        self.dev_mode = False
+        """Whether this testsuite will be ran in dev mode or not. This is
```
Comment (pmderodat) [inline e3/testsuite/__init__.py]: Typo: `will be run` (same below)

File: e3/testsuite/__init__.py [hunk 2]
```diff
@@ -168,12 +190,15 @@ def testsuite_main(self, args=None):
...
             " is kept only to keep backward compatibility of invocation with"
             " former framework (gnatpython.testdriver)")
-        parser.add_argument(
-            "--disable-cleanup",
-            dest="enable_cleanup",
-            action="store_false",
-            default=True,
-            help="disable cleanup of working space")
+
+        if self.dev_mode:
```
Comment (pmderodat) [inline e3/testsuite/__init__.py]: According to the spec above, this `--disable-cleanup` should NOT exist in dev mode, so I believe this condition should be negated.

File: e3/testsuite/__init__.py (hunk 1 above)
Comment (raph-amiard) [reply e3/testsuite/__init__.py]: Good catch, fixed !

File: e3/testsuite/__init__.py (hunk 2 above)
Comment (raph-amiard) [reply e3/testsuite/__init__.py]: Good catch, fixed

**Comment by pmderodat** [comment]
As discussed live: we now have a development mode (even though a bit different), so closing this. If you want, you can re-open a PR just for the doc additions (thanks for these!).

**Pull request closed without merging**